### PR TITLE
test: add unit tests 

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

- Adding unit tests for classes in linode_metadata/objects
- Add workflow file to run unit tests in GHA

## ✔️ How to Test

make unittest

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

**Review by lgarber-akamai — APPROVED** [review]
All tests are passing locally when running `make unittest`, nice work!

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: update packages
+        run: sudo apt-get update -y
+
+      - name: setup python 3
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      - name: install or update Python build system
+        run: python3 -m pip install -U wheel setuptools certifi pip pytest requests
```
Comment (lgarber-akamai) [inline .github/workflows/unit-tests.yml:24]: optional: We can consume the `dev` optional dependencies here.

e.g. 
```bash
python3 -m pip install ".[dev]"
```

File: Makefile [hunk 2]
```diff
@@ -65,3 +65,7 @@ autoflake:
 
 .PHONY: format
 format: black isort autoflake
+
+.PHONY: unittest
+unittest:
```
Comment (lgarber-akamai) [inline Makefile:70]: optional: Could this become `unit-test` to be consistent with https://github.com/linode/go-metadata/pull/31?

**Review by yec-akamai — APPROVED** [review]
LGTM! Tests passed locally.

File: Makefile (hunk 2 above)
Comment (ykim-1) [reply Makefile:70]: nice catch

**Pull request merged**